images/linux/scripts/installers/codeql-bundle.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+################################################################################
+##  File:  codeql-bundle.sh
+##  Desc:  Install the CodeQL CLI Bundle to the toolcache.
+################################################################################
+
+source $HELPER_SCRIPTS/install.sh
+
+# Retrieve the name of the CodeQL bundle preferred by the Action (in the format codeql-bundle-YYYYMMDD).
+codeql_bundle_name="$(curl -sSL https://raw.githubusercontent.com/github/codeql-action/main/src/defaults.json | jq -r .bundleVersion)"
+# Convert the bundle name to a version number (0.0.0-YYYYMMDD).
+codeql_bundle_version="0.0.0-${codeql_bundle_name##*-}"
+
+extraction_directory="$AGENT_TOOLSDIRECTORY/CodeQL/$codeql_bundle_version/x64"
+mkdir -p "$extraction_directory"
+
+echo "Downloading CodeQL bundle $codeql_bundle_version..."
+download_with_retries "https://github.com/github/codeql-action/releases/download/$codeql_bundle_name/codeql-bundle.tar.gz" "/tmp" "codeql-bundle.tar.gz"
+tar -xzf "/tmp/codeql-bundle.tar.gz" -C "$extraction_directory"
+
+# Touch a special file that indicates to the CodeQL Action that this bundle was baked-in to the hosted runner images.
+touch "$extraction_directory/pinned-version"
```

Comment:
What is this file `pinned-version`? Why we need it? Am I right that it is just some kind of flag file?

Sorry, I forgot to mention, you should also add one more flag to make sure that toolcache find function will find your version: https://github.com/actions/toolkit/tree/main/packages/tool-cache#find
You should create `x64.complete` file in `$AGENT_TOOLSDIRECTORY/CodeQL/$codeql_bundle_version` folder.

So final structure should be
```
- $AGENT_TOOLSDIRECTORY/CodeQL/$codeql_bundle_version
--- x64.complete
--- x64
------ <your_code_here>
```

The same for Windows

Reply:
Thanks! I've added this.

Yes, the `pinned-version` file is a file I've added at the request of my colleague @cbraynor. The purpose of it is to indicate that the CodeQL Action is running on a hosted runner and should therefore prefer the version of CodeQL in the toolcache over downloading a new version (we chose not to use this same behavior on self-hosted runners as they persist the toolcache between jobs so downloading a newer version likely makes sense).